src/network/core/os_abstraction.h

```diff
@@ -145,12 +145,16 @@ typedef unsigned long in_addr_t;
  */
 static inline bool SetNonBlocking(SOCKET d)
 {
+#ifndef __EMSCRIPTEN__
 #ifdef _WIN32
 	u_long nonblocking = 1;
 #else
 	int nonblocking = 1;
 #endif
 	return ioctlsocket(d, FIONBIO, &nonblocking) == 0;
+#else
+	return true;
```

Comment:
could roll this into the existing ifdef with an `#elif`, i think?

Reply:
I don't see how, existing ifdef ends before ioctlsocket. (early return maybe, but that could cause unreachable code warnings)